File: tests/0001_two_sum_test.cpp
#include "0001_two_sum.cpp"
#include <doctest.h>

TEST_CASE("SimpleTest")
{
    Solution solution = Solution();
    REQUIRE(solution.twoSum({2, 7, 11, 15}, 9) == std::vector{0, 1});
    REQUIRE(solution.twoSum({3, 2, 4}, 6) == std::vector{1, 2});
}

File: tests/0001_two_sum_test.py
import importlib
import pytest

module = importlib.import_module("algorithms.0001_two_sum")


def test_two_sum():
    s = module.Solution()
    assert s.twoSum(nums=[2, 7, 11, 15], target=9) == [0, 1]
    assert s.twoSum(nums=[3, 2, 4], target=6) == [1, 2]
